File: src/main/java/pl/training/module6/factory/after/ProbeService.java
package pl.training.module6.factory.after;

import java.util.Objects;

import pl.training.module6.factory.after.DeploymentProbeFactory.ProbeKind;

public final class ProbeService {
    private final DeploymentProbeFactory factory;

    public ProbeService(DeploymentProbeFactory factory) {
        this.factory = Objects.requireNonNull(factory, "factory");
    }

    public String check(ProbeKind kind, String target) {
        return factory.create(kind, target).check();
    }
}

File: typescript/src/module6/factory/after/ProbeService.ts
import { requireNonNull } from '../../../shared/requireNonNull.js';
import type { DeploymentProbeFactory, ProbeKind } from './DeploymentProbeFactory.js';

export class ProbeService {
  private readonly factory: DeploymentProbeFactory;

  constructor(factory: DeploymentProbeFactory) {
    this.factory = requireNonNull(factory, 'factory');
  }

  check(kind: ProbeKind, target: string): string {
    return this.factory.create(kind, target).check();
  }
}
